### DISCUSSION TITLE
Separate LogLevel for plugins?

### DISCUSSION THREAD

**Discussion by shoaibshakeel381** [ZiggyCreatures/FusionCache#122] [Ideas]
I looked at source code and it seems like a separate LogLevel options for plugins has not been set and everytime FusionCache instance is created it logs varikous information about found plugins. I would like to know if it would be possible to provide option to configure this? 

I would be willing to submit a PR if needed.

**Comment by jodydonetti** [comment]
Hi @shoaibshakeel381 and thanks for using FusionCache!

Can you give me a little bit more info about your scenario?
What version, OS, etc? Are you creating instances via `new FusionCache(...)` or via DI like `services.AddFusionCache(...)`? Can you give me a sample of the log output you are seeing?

Thanks!

**Reply by shoaibshakeel381** [reply]
I am using `services.AddFusionCache()` and then using DI to use `IFusionCache`. I hve registered a plugin using `services.AddSingleton<IFusionCachePlugin, CacheEventMetricsPlugin>();`

Now whenever a new instance of `IFusionCache` is created, usually on app-start, I see a log message about new plugin.

`FUSION: a plugin has been added and started (TYPE="MyNamespace.CacheEventMetricsPlugin")`

This log is coming from [Log Line](https://github.com/ZiggyCreatures/FusionCache/blob/cf8965e55d81bcf006b2d0ceb7f5620d1ed997a9/src/ZiggyCreatures.FusionCache/FusionCache.cs#L425)

Looking at all logs related to plugins, we can see that there's no plugin specfic log level. Which is what I am requesting so I can suppress such logs

**Reply by jodydonetti** [reply]
Thanks for the info, got it!

I'll put down a proposal for this change (and maybe a couple similar ones): I think it can easily be included in the new `v0.20.0` which will be released very soon.

**Reply by jodydonetti** [reply]
Hi @shoaibshakeel381 , I just created a [new issue](https://github.com/ZiggyCreatures/FusionCache/issues/128) to track the design and development of this: the implementation is already done (including some related unit tests) and is temporarily available in a [separate branch](https://github.com/ZiggyCreatures/FusionCache/tree/feature/plugins_log_levels).

The feature will be part of the new [v0.20.0](https://github.com/ZiggyCreatures/FusionCache/milestone/2), which will be released hopefully this weekend.

Hope this helps.

**Reply by shoaibshakeel381** [reply]
thanks a lot, I viewd your changes. look good 👍
